### PR TITLE
Fix progressbar artifacts within linter errors

### LINKED ISSUE #1841: Progress bar issue with lint errors
@adam-tokarski noticed this today with the progress bar feature.

When you get lint errors, they interspace the progress bar, leaving an old copy of the bar around:

Screenshot

It also makes the lints a tiny bit more difficult to read.

Is it possible to delete the progress bar lint before printing any lint error to avoid this?

### PR DESCRIPTION
**Pull request by adam-tokarski**
### Brief summary of the change made
Fixes #1841 

These changes allowed to avoid disabling progressbar when in verbose mode, which is good I guess.

### Are there any other side effects of this change that we should be aware of?
The printing (echoing/logging) was touched, so it may - but shouldn't - affect messages printed to console and/or file.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Comment by tunetheweb** [comment]
Tested this briefly and looks good to me - great work as thought we were going to have to live with it as it was!

Can you look at the coverage issues and either `# pragma: no cover` or add test cases? Then think we can merge.

**Review by tunetheweb — APPROVED** [review]

**Comment by tunetheweb** [comment]
@barrywhart / @WittierDinosaur you wanna cast your more experienced Python eyes over this before we merge?

Functionally it works a lot better anyway in my opinion.

File: src/sqlfluff/cli/commands.py [hunk 1]
```diff
@@ -62,6 +64,24 @@ def filter(self, record: logging.LogRecord) -> bool:
...
+    Without it, there were left artifacts in DEBUG mode (not sure about another ones,
+    but probably would happen somewhere).
+    """
+
+    def emit(self, record: LogRecord) -> None:
+        """Behaves like original one except uses `tqdm` to write."""
+        try:
+            msg = self.format(record)
+            tqdm.write(msg, file=self.stream)
+            self.flush()
```
Comment (barrywhart) [inline src/sqlfluff/cli/commands.py:80]: Just curious -- what happens if you omit this line?

**Review by barrywhart — APPROVED** [review]

**Comment by barrywhart** [comment]
The code changes look great! Haven't tested it, but @tunetheweb did, so I feel good about this PR.

File: src/sqlfluff/cli/commands.py (hunk 1 above)
Comment (adam-tokarski) [reply src/sqlfluff/cli/commands.py:80]: Seems to be working as far I can tell, but it is in original `emit`, so I guess this is for some reason. So it safer just to put it there.

**Pull request merged**